src/backend/distributed/executor/local_executor.c

```diff
@@ -589,6 +590,47 @@ ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
 }
 
 
+/*
+ * SetLocalExecutionStatus sets the local execution status to
+ * the given status, it errors if the transition is not possible from the
+ * current status.
+ */
+void
+SetLocalExecutionStatus(LocalExecutionStatus newStatus)
+{
+	EnsureTransitionPossible(CurrentLocalExecutionStatus, newStatus);
+
+	CurrentLocalExecutionStatus = newStatus;
+}
+
+
+/*
+ * EnsureTransitionPossible errors if we cannot switch to the 'to' status
+ * from the 'from' status.
+ */
+static void
+EnsureTransitionPossible(LocalExecutionStatus from, LocalExecutionStatus
+						 to)
+{
+	if (from == LOCAL_EXECUTION_REQUIRED && to == LOCAL_EXECUTION_DISABLED)
+	{
+		ereport(ERROR,
+				(errmsg(
+					 "cannot switch local execution status from local execution required "
+					 "to local execution disabled since it can cause "
+					 "visibility problems in the current transaction")));
```

Comment:
Is there a way to exercise either of these code paths?

Reply:
if there is then there is a bug. So not sure how to test this.

Reply:
There was one test failure hitting this path with insert..select inside a transaction block, but after #3714 it is resolved